src/connector/src/parser/avro/parser.rs

```diff
@@ -248,35 +302,23 @@ impl AvroParser {
                 Op::Insert => writer.insert(fill),
                 Op::Delete => writer.delete(fill),
             }
-        } else if self.upsert_primary_key_column_name.is_some()
-            && matches!(op, Op::Delete)
-            && matches!(
-                avro_value,
-                Value::Boolean(_)
-                    | Value::String(_)
-                    | Value::Int(_)
-                    | Value::Long(_)
-                    | Value::Float(_)
-                    | Value::Decimal(_)
-                    | Value::Date(_)
-                    | Value::TimestampMillis(_)
-                    | Value::TimestampMicros(_)
-                    | Value::Duration(_)
-            )
-        {
+        } else if self.upsert_primary_key_column_name.is_some() && matches!(op, Op::Delete) {
             writer.delete(|desc| {
                 if &desc.name != self.upsert_primary_key_column_name.as_ref().unwrap() {
                     Ok(None)
                 } else {
-                    from_avro_value(avro_value.clone(), self.key_schema.as_deref().unwrap())
-                        .map_err(|e| {
-                            tracing::error!(
-                                "failed to process value ({}): {}",
-                                String::from_utf8_lossy(&payload),
-                                e
-                            );
+                    from_avro_value(
+                        avro_key.as_ref().unwrap().clone(),
+                        self.key_schema.as_deref().unwrap(),
+                    )
+                    .map_err(|e| {
+                        tracing::error!(
+                            "failed to process value ({}): {}",
+                            String::from_utf8_lossy(&payload),
                             e
-                        })
+                        );
+                        e
+                    })
                 }
             })
         } else {
```

Comment:
Can we also include the type of `avro_value`, `self.upsert_primary_key_column_name` and `op` in the error message below?